docs/source/argument_check.rst

```diff
@@ -0,0 +1,607 @@
+                                                        const void*       alpha,
+                                                        const void*       a,
+                                                        rocblas_datatype  a_type,
+                                                        rocblas_int       lda,
+                                                        rocblas_stride    stride_a,
+                                                        const void*       b,
+                                                        rocblas_datatype  b_type,
+                                                        rocblas_int       ldb,
+                                                        rocblas_stride    stride_b,
+                                                        const void*       beta,
+                                                        const void*       c,
+                                                        rocblas_datatype  c_type,
+                                                        rocblas_int       ldc,
+                                                        rocblas_stride    stride_c,
+                                                        void*             d,
+                                                        rocblas_datatype  d_type,
+                                                        rocblas_int       ldd,
+                                                        rocblas_stride    stride_d,
+                                                        rocblas_int       batch_count,
+                                                        rocblas_datatype  compute_type,
+                                                        rocblas_gemm_algo algo,
+                                                        int32_t           solution_index,
+                                                        uint32_t          flags)
+    {
+        // handle must not be null pointers for logging
+        if(!handle)
+            return rocblas_status_invalid_handle;
+
+        auto layer_mode = handle->layer_mode;
+  //----snip---logging---does---not---dereferences---null---pointer-------------------------------
+
+       // sizes must not be negative
+        if(m < 0 || n < 0 || k < 0 || batch_count < 0)
+            return rocblas_status_invalid_size;
+
+        rocblas_int num_rows_a = trans_a == rocblas_operation_none ? m : k;
+        rocblas_int num_rows_b = trans_b == rocblas_operation_none ? k : n;
+        rocblas_int num_rows_c = m;
+        rocblas_int num_rows_d = m;
+
+        // leading dimensions must be valid
+        if(num_rows_a > lda || num_rows_b > ldb || num_rows_c > ldc || num_rows_d > ldd)
+            return rocblas_status_invalid_size;
+
+        // quick return m,n,k equal to 0 is valid in BLAS
+        if(!m || !n || !k || !batch_count)
+            return rocblas_status_success;
+
+       // --- on host if rocblas_pointer_mode_host, on device if rocblas_pointer_mode_device ---
```

Comment:
Only host pointer data can be queried for values that allow early success for identity operations.   Think the next lines should be alpha not !alpha ?